source/simulator/entropy_viscosity.cc

```diff
@@ -599,14 +596,32 @@ namespace aspect
                  parameters.composition_degree
                 );
             const double h = cell->diameter();
+            const double eps = max_conductivity_on_cell;
 
-            // Compute tau for SUPG
-            viscosity_per_cell[cell->active_cell_index()] = (h/fe_order)/(2*norm_of_advection_term);
+            // SUPG parameter design from "On Discontinuity-Capturing Methods
+            // for Convection-Diffusion Equations" by Volker John and Petr
+            // Knobloch. Also see deal.II step-63:
+            // delta_k = h / (2 \|u\|) * (coth(Pe) - 1/Pe)
 
-            if (max_conductivity_on_cell>1e-8)
-              viscosity_per_cell[cell->active_cell_index()] *= (1.0/tanh(norm_of_advection_term*(h/fe_order)/(2.0*max_conductivity_on_cell))
-                                                                - 1.0/(norm_of_advection_term*(h/fe_order)/(2.0*max_conductivity_on_cell)));
+            // Pe = \| u \| h/(2 p eps)
+            const double peclet_times_eps = norm_of_advection_term * h / (2.0 * fe_order);
 
+            // Instead of Pe < 1, we check Pe*eps < eps as eps can be ==0:
+            if (peclet_times_eps==0.0 || peclet_times_eps < eps)
+              {
+                // Diffusion dominant case, no stabilization needed:
+                viscosity_per_cell[cell->active_cell_index()] = 0.0;
+              }
+            else
+              {
+                // To avoid a division by zero, increase eps slightly. The actual value is not
+                // important, as long as the result is still a valid number. Note that this
+                // is only important if \|u\| and eps are zero.
+                const double peclet = peclet_times_eps / (eps + 1e-100);
```

Comment:
Do we need `eps+1e-100` here? If `eps=0`, the `peclet_times_eps==0.0` above will be true, correct? I guess it doesn't really matter though...